prow/tide/tide.go

```diff
@@ -1258,7 +1278,7 @@ func (c *Controller) takeAction(sp subpool, batchPending, successes, pendings, m
 	// Do not merge PRs while waiting for a batch to complete. We don't want to
 	// invalidate the old batch result.
 	if len(successes) > 0 && len(batchPending) == 0 {
-		if ok, pr := pickSmallestPassingNumber(sp.log, c.ghc, successes, sp.cc); ok {
+		if ok, pr := pickHighestPriorityPR(sp.log, c.ghc, successes, sp.cc, isPassingTests, c.config().Tide.Priority); ok {
```

Comment:
Why the added isPassingTests? The fact that this is in `successes` implies that it has passing tests I think

Reply:
Look at my comment https://github.com/kubernetes/test-infra/pull/20468/files#r559936185